lib/puppet/type/file/source.rb

```diff
@@ -286,8 +285,10 @@ def each_chunk_from
 
     def chunk_file_from_disk
       File.open(full_path, "rb") do |src|
-        while chunk = src.read(8192)
+        chunk = src.read(8192)
+        while chunk
           yield chunk
+          chunk = src.read(8192)
```

Comment:
Just curious why this one got broken out instead of an ignore comment.

Reply:
I don't know what you're referring here?

Reply:
For example, in ext/yaml_nodes.rb above,
```
while parent = info["parent"]
```
was changed to
```
while parent = info["parent"] #rubocop:disable Lint/AssignmentInCondition
```
So I was curious about why `#rubocop:disable Lint/AssignmentInCondition` was not used on this while assignment.